application/src/main/java/org/togetherjava/tjbot/commands/moderation/BanCommand.java

```diff
@@ -72,9 +76,9 @@ public BanCommand() {
     }
 
     @SuppressWarnings("MethodWithTooManyParameters")
-    private static RestAction<InteractionHook> banUserFlow(@NotNull User target,
-            @NotNull Member author, @NotNull String reason, int deleteHistoryDays,
-            @NotNull Guild guild, @NotNull SlashCommandEvent event) {
+    private RestAction<InteractionHook> banUserFlow(@NotNull User target, @NotNull Member author,
+            @NotNull String reason, int deleteHistoryDays, @NotNull Guild guild,
+            @NotNull SlashCommandEvent event) {
```

Comment:
It is checked by the system though, no? Its new to me that you should actually trigger a NPE explicitly if you already use `@NotNull` annotations.

Reply:
>>> Its new to me that you should actually trigger a NPE explicitly if you already use @NotNull annotations

Al lot of JDA methods have @NotNull but we still check it, so sounds inconsistent to only check it on their side.

Besides that, you can use a plugin which generates requireNonNull for you, this works in IntelliJ but not sure whenever it works outside of that.

Reply:
I just havent seen this yet, people explicitly using `requireNonNull` (or other triggers for NPE) on stuff that is marked `@NotNull` already. @marko-radosavljevic , @I-Al-Istannen , do you know more here about convention?

Reply:
It has to do with your fail fast principle.

Fail fast, don't wait till you get a random NPE.

If you accept null, it should be `@Nullable` NOT `@NotNull`

Reply:
> I don't think I understand what you mean, or well it seems like there's some kind of a miscommunication.

I do understand you. I just was under the impression that it also has/does the checks without plugin. Or that its so widely supported across IDEs by default already that people generally dont check it extra again by convention.

I am just asking about the convention right now.

Reply:
> Like, simply passing a null object to a method annoted with @NotNull already triggers a NPE even if you do not check it yourself in the code, the check is "built-in".

Have you tested this? It's something intellIj does during compilation, if you enable it. Which has several issues, we are building it and running it from gradle, also what about other IDEs, etc.

> When you compile your project, the IDE adds assertions to all methods and parameters annotated with the @NotNull annotation. The assertions will fail if null is passed in code where @NotNull is expected. You can disable this option and configure the list of annotations in the Settings/Preferences dialog Ctrl+Alt+S. Go to Build, Execution, Deployment | Compiler. 

That said, our static checkers support this annotation, so sonarlint fails the build with: [[java:S2637] "@NonNull" values should not be set to null](https://sonarcloud.io/organizations/togetherjava/rules?open=java%3AS2637&rule_key=java%3AS2637).

Also, not sure how other IDEs support this annotation, for example does this work:
![image](https://user-images.githubusercontent.com/30927961/145245360-952e38bd-358a-4d84-9dca-ffd238f2b0f5.png)

All in all, not sure. We have a static checking, but I would like to have a runtime check as well. If there is a plugin that does this for gradle, it would be fine with me.

Reply:
Sounds reasonable. I guess I will just add the checks and keep it in mind for next time. Good spot, thanks.

Reply:
I will trust the call-site for private methods though. Otherwise we have to add like 100 checks to every method in the code base, not just the ones you flagged. So for everything that is `private`, I wont start checking everything.